packages/react/src/Hero/Hero.tsx

```diff
@@ -15,36 +19,63 @@ export type HeroProps = BaseProps<HTMLElement> & {
 
 const Root = forwardRef<HTMLElement, PropsWithChildren<HeroProps>>(
   ({className, align = 'start', children, ...rest}, ref) => {
-    const {HeroActions, HeroChildren} = useMemo(
+    const {HeroActions, HeroChildren, HeroImageChild} = useMemo(
       () =>
         React.Children.toArray(children).reduce<{
           HeroActions: React.ReactElement[]
+          HeroImageChild?: React.ReactElement
           HeroChildren: React.ReactElement[]
         }>(
           (acc, child) => {
             if (React.isValidElement(child)) {
               if (child.type === HeroPrimaryAction || child.type === HeroSecondaryAction) {
                 acc.HeroActions.push(child)
+              } else if (child.type === HeroImage) {
+                acc.HeroImageChild = child
               } else {
                 acc.HeroChildren.push(child)
               }
             }
             return acc
           },
-          {HeroActions: [], HeroChildren: []},
+          {HeroActions: [], HeroChildren: [], HeroImageChild: undefined},
         ),
       [children],
     )
 
+    const imagePosition = HeroImageChild?.props?.imagePosition || 'block-end'
+
+    const heroLayoutClass = HeroImageChild ? styles['Hero--layout-image'] : styles['Hero--layout-default']
+
     return (
       <section
-        className={clsx(styles.Hero, styles[`Hero--align-${align}`], className)}
+        className={clsx(
+          styles.Hero,
+          imagePosition !== 'inline-end' && styles[`Hero--align-${align}`],
+          heroLayoutClass,
+          HeroImageChild && styles[`Hero--image-pos-${imagePosition}`],
+        )}
         ref={ref}
         aria-labelledby="hero-section-brand-heading"
         {...rest}
       >
-        {HeroChildren}
-        <div className={styles['Hero-actions']}>{HeroActions}</div>
+        <Grid fullWidth className={clsx(styles['Hero-grid'], styles[`Hero-grid--${imagePosition}`])}>
+          <Grid.Column span={{medium: HeroImageChild && imagePosition === 'inline-end' ? 6 : 12}}>
+            <Stack
+              direction="vertical"
+              gap="none"
+              padding="none"
+              alignItems={imagePosition === 'inline-end' ? undefined : align === 'start' ? 'flex-start' : 'center'}
```

Comment:
For `inline-end` we also need to set `align-items` to `flex-start`, otherwise it will default to `stretch` and make the Label span the full width of the Stack container.

<img width="846" alt="screenshot-dDkJ87H4-000602@2x" src="https://github.com/primer/brand/assets/175638/91ac670c-a3aa-4997-aabf-05bc15d34bfa">

Reply:
👍 fixed